File: backend/src/models/entity.py
from enum import StrEnum
from typing import Any

from pydantic import BaseModel, Field


class EntityType(StrEnum):
    PERSON = "person"
    ORG = "org"
    OPENSOURCE = "opensource"
    MEDIA = "media"


CATEGORY_LABELS = {
    "core_leaders": "🧠 核心领袖",
    "top_brands": "🏢 顶级厂牌",
    "opensource_geeks": "🛠️ 开源与极客",
    "industry_intel": "📰 行业内参",
}


class Entity(BaseModel):
    id: str
    name: str
    avatar: str = ""
    identity_tag: str = ""
    type: EntityType
    bio: str = ""
    activity_tags: list[str] = Field(default_factory=list)
    influence: dict[str, str] = Field(default_factory=dict)
    recent_highlights: list[str] = Field(default_factory=list)
    scrape_sources: list[dict[str, Any]] = Field(default_factory=list)
    category: str = "core_leaders"


class Post(BaseModel):
    id: str
    entity_id: str
    source_type: str = ""
    source_url: str = ""
    original_text: str = ""
    summary_zh: str = ""
    insight_zh: str = ""
    interpretation_zh: str = ""
    translation_zh: str = ""
    published_at: str = ""
    created_at: str = ""
    is_trending: bool = False
    trending_source: str = ""

File: client/lib/models/entity.dart
enum EntityType { person, org, opensource, media }

class Entity {
  final String id;
  final String name;
  final String avatar;
  final String identityTag;
  final EntityType type;
  final String bio;
  final List<String> activityTags;
  final Map<String, String> influence;
  final List<String> recentHighlights;
  final String category;

  Entity({
    required this.id,
    required this.name,
    this.avatar = '',
    this.identityTag = '',
    required this.type,
    this.bio = '',
    this.activityTags = const [],
    this.influence = const {},
    this.recentHighlights = const [],
    this.category = 'core_leaders',
  });

  factory Entity.fromJson(Map<String, dynamic> json) {
    return Entity(
      id: json['id'] as String,
      name: json['name'] as String,
      avatar: json['avatar'] as String? ?? '',
      identityTag: json['identity_tag'] as String? ?? '',
      type: EntityType.values.firstWhere(
        (e) => e.name == json['type'],
        orElse: () => EntityType.person,
      ),
      bio: json['bio'] as String? ?? '',
      activityTags: (json['activity_tags'] as List<dynamic>?)
              ?.map((e) => e as String)
              .toList() ??
          [],
      influence: (json['influence'] as Map<String, dynamic>?)
              ?.map((k, v) => MapEntry(k, v.toString())) ??
          {},
      recentHighlights: (json['recent_highlights'] as List<dynamic>?)
              ?.map((e) => e as String)
              .toList() ??
          [],
      category: json['category'] as String? ?? 'core_leaders',
    );
  }
}
